server/src/main/java/org/apache/iotdb/db/engine/storagegroup/StorageGroupProcessor.java

```diff
@@ -1974,6 +1945,10 @@ private int findInsertionPosition(TsFileResource newTsFileResource, long newFile
         continue;
       }
 
+      if (!localFile.isClosed() && localFile.getProcessor() != null) {
```

Comment:
So we have to close all files in a SG if we want to load a new TsFile...

Reply:
Sure, or the next line will not work.